src/test/lrsql/admin/protocol_test.clj

```diff
@@ -155,7 +187,79 @@
+                                  {} ifis))]
+        (lrsp/-store-statements lrs auth-ident stmts [])
+
+        (is (every? #(> % 0) (vals (get-stmt-#s))))
+        (doseq [ifi ifis]
+          (adp/-delete-actor lrs {:actor-ifi ifi}))
+        (is (every? #(= % 0) (vals (get-stmt-#s))))))
+    (testing "actor delete and related statements"
+      (let [stmt->ifi #(ua/actor->ifi (% "actor"))
+            count-of-actor (fn [actor-ifi] (-> (lrsp/-get-statements lrs auth-ident {:actor-ifi actor-ifi} []) :statement-result :statements count))
+            child-ifi (stmt->ifi (stmt-3 "object"))
+            parent-ifi (stmt->ifi stmt-3)]
+        (testing "delete-actor correctly deletes statements that are parent to actor (sub)statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor parent-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi})
+          (is (zero? (count-of-actor parent-ifi))) ;
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi}))
+        (testing "delete-actor correctly deletes substatements that are child to actor statements"
+          (lrsp/-store-statements lrs auth-ident [stmt-3] [])
+          (is (= 1 (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi parent-ifi})
+          (is (zero? (count-of-actor child-ifi)))
+          (adp/-delete-actor lrs {:actor-ifi child-ifi}))
+        (testing "for StatementRefs, delete-actor deletes statement->actor relationships but leaves statements by another actor untouched"
+          (let [[ifi-0 ifi-2] (mapv stmt->ifi [stmt-0 stmt-2])]
+            (lrsp/-store-statements lrs auth-ident [stmt-0 stmt-2] [])
+            (is (= 2 (count-of-actor ifi-0)))
+            (is (= 2 (count-of-actor ifi-2)))
+            (adp/-delete-actor lrs {:actor-ifi ifi-0})
+            (is (= 1 (count-of-actor ifi-0)))
+            (is (= 1 (count-of-actor ifi-2)))
+            (adp/-delete-actor lrs {:actor-ifi ifi-2})
+            (is (zero? (count-of-actor ifi-0)))
+            (is (zero? (count-of-actor ifi-2)))))))
+
+    (testing "delete-actor handles dependent tables: state_document"
+      (let [ifi (ua/actor->ifi (:agent lrst/state-id-params))
+            ds (-> lrs :connection :conn-pool)]
+      (lrsp/-set-document lrs auth-ident lrst/state-id-params lrst/state-doc-1 true)
+      (adp/-delete-actor lrs {:actor-ifi ifi})
+      (is (empty? (jdbc/execute! ds ["select * from state_document where agent_ifi  = ?" ifi])))))
+
+    (testing "delete-actor handles dependent tables: agent_profile_document"
+      (let [ifi (ua/actor->ifi (:agent lrst/agent-prof-id-params))
+            ds (-> lrs :connection :conn-pool)]
+        (lrsp/-set-document lrs auth-ident lrst/agent-prof-id-params lrst/agent-prof-doc true)
+        (adp/-delete-actor lrs {:actor-ifi ifi})
+        (is (empty? (jdbc/execute! ds ["select * from agent_profile_document where agent_ifi  = ?" ifi])))))
```

Comment:
You can use the [get-document](https://github.com/yetanalytics/lrsql/blob/SQL-201/src/main/lrsql/system/lrs.clj#L169) protocol function instead of hand-writing a query. You would have to specify a `:stateId` to query the state document, and you would have to specify a `:profileId` and `:agent` to query the agent profile document.

Reply:
switched to use of `get-document`